src/map/script.c

```diff
@@ -17351,41 +17351,51 @@ BUILDIN(petstat)
 
 BUILDIN(callshop)
 {
-	struct npc_data *nd;
-	const char *shopname;
-	int flag = 0;
+	const char *shopname = script_getstr(st, 2);
+	struct npc_data *nd = npc->name2id(shopname);
+	int flag = script_hasdata(st, 3) ? script_getnum(st, 3) : 0;
 	struct map_session_data *sd = script->rid2sd(st);
 
-	if (sd == NULL)
+	if (sd == NULL) {
+		ShowWarning("script:callshop: no player attached.\n");
+		script_pushint(st, 0);
 		return true;
-	shopname = script_getstr(st, 2);
-	if( script_hasdata(st,3) )
-		flag = script_getnum(st,3);
-	nd = npc->name2id(shopname);
-	if( !nd || nd->bl.type != BL_NPC || (nd->subtype != SHOP && nd->subtype != CASHSHOP) )
-	{
-		ShowError("buildin_callshop: Shop [%s] not found (or NPC is not shop type)\n", shopname);
-		script_pushint(st,0);
+	}
+
+
+	if (nd == NULL || nd->bl.type != BL_NPC || (nd->subtype != SHOP && nd->subtype != CASHSHOP &&
+		!(nd->subtype == SCRIPT && nd->u.scr.shop != NULL && nd->u.scr.shop->items > 0))) {
```

Comment:
Spliting this check would be better with a proper error message see `openshop` 

 `ShowWarning("buildin_openshop: trying to open without any items!\n");`

Reply:
`script:callshop: trader '%s' has no shop list.` is the same thing